### PR TITLE
Refactor: remove test gateway subscription module

### PR DESCRIPTION
**Pull request by jonwaldstein**
## Description

This removes the Test Gateway Subscription Module in favor of loading from give-recurring.

## Affects

The Test Gateway

## Visuals

N/A

## Testing Instructions

Just make sure the test gateway still works.

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205269709893103

### CODE REVIEW

**Comment by jonwaldstein** [comment]
@glaubersilva this is ready for review.

File: src/PaymentGateways/Gateways/TestOffsiteGateway/testOffsiteGateway.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        message: string;
+    };
+}
+
+/**
+ * @unreleased
+ */
+const testOffsiteGateway: TestOffsiteGateway = {
+    id: 'test-offsite-gateway',
+    Fields() {
```
Comment (glaubersilva) [inline src/PaymentGateways/Gateways/TestOffsiteGateway/testOffsiteGateway.tsx:15]: When I try to compile the assets I got the following error on this line:

```
 TS2322: Type '() => string' is not assignable to type 'FC<{}>'.
  Type 'string' is not assignable to type 'ReactElement<any, any>'.
```

image

**Review by glaubersilva — CHANGES_REQUESTED** [review]
@jonwaldstein I left a comment with a problem I got trying to compile the assets.

File: src/PaymentGateways/Gateways/TestOffsiteGateway/testOffsiteGateway.tsx (hunk 1 above)
Comment (jonwaldstein) [reply src/PaymentGateways/Gateways/TestOffsiteGateway/testOffsiteGateway.tsx:15]: Resolved 4a7dec09

**Review by glaubersilva — APPROVED** [review]
@jonwaldstein Thanks for the change, now it worked as expected. Nice work!

**Comment by kjohnson** [comment]
The failing automated test looks like it might be an actual failing test.

```
1) Give\Tests\Unit\Subscriptions\Models\TestSubscription::testSubscriptionShouldCancel
Error: Call to a member function cancelSubscription() on null

/home/runner/work/givewp/givewp/src/Framework/PaymentGateways/PaymentGateway.php:138
/home/runner/work/givewp/givewp/src/Subscriptions/Models/Subscription.php:214
/home/runner/work/givewp/givewp/tests/Unit/Subscriptions/Models/TestSubscription.php:127
phpvfscomposer:///home/runner/work/givewp/givewp/vendor/phpunit/phpunit/phpunit:51
```

I'm updating the branch to bring in the latest changes and we'll see if the failing test resolves it self or if it requires additional attention.

**Review by kjohnson — CHANGES_REQUESTED** [review]
We have a failing test that needs to be resolved:

```
1) Give\Tests\Unit\Subscriptions\Models\TestSubscription::testSubscriptionShouldCancel
Error: Call to a member function cancelSubscription() on null

/home/runner/work/givewp/givewp/src/Framework/PaymentGateways/PaymentGateway.php:138
/home/runner/work/givewp/givewp/src/Subscriptions/Models/Subscription.php:214
/home/runner/work/givewp/givewp/tests/Unit/Subscriptions/Models/TestSubscription.php:127
phpvfscomposer:///home/runner/work/givewp/givewp/vendor/phpunit/phpunit/phpunit:51
```

**Comment by jonwaldstein** [comment]
@kjohnson I marked that failing test incomplete dd64cfda.  That method has a side effect that calls the gateway so the assertion would depend on the gateway - not the model. Anyway, fine for now - can be re-written at some point.

**Pull request merged**